app.ts

```diff
@@ -214,22 +216,24 @@ props.initialize(configDir, propHierarchy);
 const ceProps = props.propsFor('compiler-explorer');
 defArgs.wantedLanguages = ceProps('restrictToLanguages', defArgs.wantedLanguages);
 
-let languages = allLanguages;
-if (defArgs.wantedLanguages) {
-    const filteredLangs = {};
-    const passedLangs = defArgs.wantedLanguages.split(',');
-    for (const wantedLang of passedLangs) {
-        for (const langId in languages) {
-            const lang = languages[langId];
-            if (lang.id === wantedLang || lang.name === wantedLang || lang.alias === wantedLang) {
-                filteredLangs[lang.id] = lang;
+const languages = (() => {
+    if (defArgs.wantedLanguages) {
+        const filteredLangs: Partial<Record<LanguageKey, Language>> = {};
+        const passedLangs = defArgs.wantedLanguages.split(',');
+        for (const wantedLang of passedLangs) {
+            for (const lang of Object.values(allLanguages)) {
+                if (lang.id === wantedLang || lang.name === wantedLang || lang.alias === wantedLang) {
+                    filteredLangs[lang.id] = lang;
+                }
             }
         }
+        // Always keep cmake for IDE mode, just in case
+        filteredLangs[allLanguages.cmake.id] = allLanguages.cmake;
+        return filteredLangs;
+    } else {
+        return allLanguages;
     }
-    // Always keep cmake for IDE mode, just in case
-    filteredLangs[languages.cmake.id] = languages.cmake;
-    languages = filteredLangs;
-}
+})();
```

Comment:
Seems good, optionally make it a function and pass allLanguages into it instead of using an IIFE

Reply:
That might be a good idea, a number of other constants in the file are also initialized with IIFE's so I may leave this for now and we could refactor stuff later